Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System;
+
+namespace BootCamp.Chapter
+{
+    class Lesson4
+    {
+        public static void Demo()
+        {
+            bool calculateAgain = true;
+
+            while (calculateAgain)
+            {
+                Console.WriteLine("BMI Calculator");
+
+                string name = PromptString("Enter your name: ");
+                int age = PromptInt("Enter your age: ");
+                float weight = PromptFloat("Enter your weight(kg): ");
+                float height = PromptFloat("Enter your height(cm): ");
+
+                Console.WriteLine(name + " is " + age + " years old, his weight is " + weight + " kg and his height is " + height + " cm.");
+
+                float BMI = CalculateBmi(weight, height / 100);
+
+                Console.WriteLine("Your BMI is " + BMI);
+
+                WeightType(BMI);
+
+                calculateAgain = ResponseValidation();
+            }
+
+            Console.WriteLine("Exiting...");
+
+        }
+
+        public static string PromptString(string message)
+        {
+            Console.Write(message);
+            var input = Console.ReadLine();
+
+            if (string.IsNullOrEmpty(input))
+            {
+                Console.WriteLine("Name cannot be empty.");
+                return "-";
+            }
+
+            return input;
+        }
+
+        public static int PromptInt(string message)
+        {
+            Console.Write(message);
+            var input = Console.ReadLine();
+
+            if (string.IsNullOrEmpty(input))
+            {
+                return 0;
+            }
+
+            if(!int.TryParse(input, out int output))
+            {
+                Console.WriteLine("\"" + input + "\" is not a valid number.");
+                return -1;
+            }
+            return output;
+        }
+
+        public static float PromptFloat(string message)
+        {
+            Console.Write(message);
+            var input = Console.ReadLine();
+
+            if (string.IsNullOrEmpty(input))
+            {
+                return 0;
+            }
+
+            if (!float.TryParse(input, out float output))
+            {
+                Console.WriteLine("\"" + input + "\" is not a valid number.");
+                return -1;
+            }
+
+            return output;
+        }
+
+        public static float CalculateBmi(float weight, float height)
+        {
+            if(weight <= 0 && height <= 0)
+            {
+                Console.WriteLine("Failed calculating BMI. Reason:");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Define a constant instead of using this literal 'Failed calculating BMI. Reason:' 4 times.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885366)